server/accounts_test.go

```diff
@@ -0,0 +1,835 @@
+		if u.Username == "derek" {
+			if u.Account != natsAcc {
+				t.Fatalf("Expected to see the 'nats.io' account, but received %+v", u.Account)
+			}
+		}
+	}
+}
+
+func TestAccountParseConfigDuplicateUsers(t *testing.T) {
+	confFileName := createConfFile(t, []byte(`
+    accounts {
+      synadia {
+        users = [
+          {user: alice, password: foo}
+          {user: bob, password: bar}
+        ]
+      }
+      nats.io {
+        users = [
+          {user: alice, password: bar}
+        ]
+      }
+    }
+    `))
+	defer os.Remove(confFileName)
+	_, err := ProcessConfigFile(confFileName)
+	if err == nil {
+		t.Fatalf("Expected an error with double user entries")
+	}
+}
+
+func TestAccountParseConfigImportsExports(t *testing.T) {
+	opts, err := ProcessConfigFile("./configs/accounts.conf")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if la := len(opts.Accounts); la != 3 {
+		t.Fatalf("Expected to see 3 accounts in opts, got %d", la)
+	}
+	if lu := len(opts.Nkeys); lu != 4 {
+		t.Fatalf("Expected 4 total Nkey users, got %d", lu)
+	}
+	if lu := len(opts.Users); lu != 0 {
+		t.Fatalf("Expected no Users, got %d", lu)
+	}
+	var natsAcc, synAcc *Account
+	for _, acc := range opts.Accounts {
+		if acc.Name == "nats.io" {
+			natsAcc = acc
+		} else if acc.Name == "synadia" {
+			synAcc = acc
+		}
+	}
+	if natsAcc == nil {
+		t.Fatalf("Error retrieving account for 'nats.io'")
+	}
+	if natsAcc.Nkey != "AB5UKNPVHDWBP5WODG742274I3OGY5FM3CBIFCYI4OFEH7Y23GNZPXFE" {
+		t.Fatalf("Expected nats account to have an nkey, got %q\n", natsAcc.Nkey)
+	}
+	// Check user assigned to the correct account.
+	for _, nk := range opts.Nkeys {
+		if nk.Nkey == "UBRYMDSRTC6AVJL6USKKS3FIOE466GMEU67PZDGOWYSYHWA7GSKO42VW" {
+			if nk.Account != natsAcc {
+				t.Fatalf("Expected user to be associated with natsAcc, got %q\n", nk.Account.Name)
+			}
+			break
+		}
+	}
+
+	// Now check for the imports and exports of streams and services.
+	if lis := len(natsAcc.imports.streams); lis != 2 {
+		t.Fatalf("Expected 2 imported streams, got %d\n", lis)
+	}
+	if lis := len(natsAcc.imports.services); lis != 1 {
+		t.Fatalf("Expected 1 imported service, got %d\n", lis)
+	}
+	if les := len(natsAcc.exports.services); les != 1 {
+		t.Fatalf("Expected 1 exported service, got %d\n", les)
+	}
+	if les := len(natsAcc.exports.streams); les != 0 {
+		t.Fatalf("Expected no exported streams, got %d\n", les)
+	}
+
+	if synAcc == nil {
+		t.Fatalf("Error retrieving account for 'synadia'")
+	}
+
+	if lis := len(synAcc.imports.streams); lis != 0 {
+		t.Fatalf("Expected no imported streams, got %d\n", lis)
+	}
+	if lis := len(synAcc.imports.services); lis != 1 {
+		t.Fatalf("Expected 1 imported service, got %d\n", lis)
+	}
+	if les := len(synAcc.exports.services); les != 2 {
+		t.Fatalf("Expected 2 exported service, got %d\n", les)
+	}
+	if les := len(synAcc.exports.streams); les != 2 {
+		t.Fatalf("Expected 2 exported streams, got %d\n", les)
+	}
+}
+
+func TestImportExportConfigFailures(t *testing.T) {
```

Comment:
In this test, please check that returned error contains string that you expect. I noticed that some errors are just malformed config (say expect map got string) and not at all what is the expected error.

Reply:
ok will take a look. This errors from the config parsing?